functions/src/googleform-hook.ts

```diff
@@ -0,0 +1,115 @@
+import * as functions from 'firebase-functions'
+import * as admin from 'firebase-admin'
+
+import { Path } from './models/Path'
+import { StepType } from './models/StepType'
+import Logger from './utils/logger'
+
+function pathLabelToPathDocumentId(pathLabel: string): Path | null {
+  switch (pathLabel) {
+    case 'Data Science & Analytics':
+      return Path.dataScienceAnalytics
+    case 'Product Design':
+      return Path.productDesign
+    case 'Product Management':
+      return Path.productManagement
+    case 'Software Engineering':
+      return Path.softwareEngineering
+    default:
+      return null
+  }
+}
+
+async function googleformWebhook(
+  req: functions.Request,
+  res: functions.Response
+) {
+  if (req.method !== 'POST') {
+    Logger.warn(
+      `[405] METHOD NOT ALLOWED: Typeform webhook called with method: ${
+        req.method
+      }, body: ${JSON.stringify(req.body)}, headers: ${JSON.stringify(
+        req.headers
+      )}`
+    )
+    res.status(405).json({
+      status: 405,
+      message: `${req.method} is not allowed`,
+    })
+    return
+  }
+  try {
+    const { body } = req
+    const submittedAt = body.submitted_at
+    const email = body.email
+    const submissionUrl = body.submission_url
+    const secretKey = body.secret_key
+    const chosenPath = pathLabelToPathDocumentId(body.path)
+
+    // TODO: move secret key to env
+    if (
+      secretKey !== 'a6d6901d8b2bfdb265fd0b0b234bda75d9e1ef0b5b1627722294c039'
```

Comment:
kwkwkw masbro, silahkan ganti secret key dan jangan lupa dimasukkan ke env

Reply:
kuganti ke functions.config().googleformHook.secret_key , tapi ini untested karena not sure gimana set nya wkwkwk

Reply:
https://firebase.google.com/docs/functions/config-env
Coba set config nya dulu terus deploy ke prod win. Kalo udah oke letss merge